### PR TITLE
Rc2

### PR DESCRIPTION
**Pull request by jonspock**

### CODE REVIEW

**Comment by PIVX-Project** [comment]
ACK

**Comment by rejectedpromise** [comment]
ACK

File: src/masternodeman.cpp [hunk 1]
```diff
@@ -811,7 +811,7 @@ void CMasternodeMan::ProcessMessage(CNode* pfrom, std::string& strCommand, CData
                     int64_t t = (*i).second;
                     if (GetTime() < t) {
                         Misbehaving(pfrom->GetId(), 34);
-                        LogPrintf("dseg - peer already asked me for the list\n");
+                        LogPrint("masternode","dseg - peer already asked me for the list\n");
```
Comment (Fuzzbawls) [inline src/masternodeman.cpp]: post-merge: This should have been left as `LogPrintf()` as it results in a `Misbehaving` message.

File: src/masternodeman.cpp [hunk 2]
```diff
@@ -877,7 +877,7 @@ void CMasternodeMan::ProcessMessage(CNode* pfrom, std::string& strCommand, CData
 
         // make sure signature isn't in the future (past is OK)
         if (sigTime > GetAdjustedTime() + 60 * 60) {
-            LogPrintf("dsee - Signature rejected, too far into the future %s\n", vin.prevout.hash.ToString());
+            LogPrint("masternode","dsee - Signature rejected, too far into the future %s\n", vin.prevout.hash.ToString());
```
Comment (Fuzzbawls) [inline src/masternodeman.cpp]: post-merge: This should have been left as `LogPrintf()` as it results in a `Misbehaving` message.

File: src/masternodeman.cpp [hunk 3]
```diff
@@ -888,7 +888,7 @@ void CMasternodeMan::ProcessMessage(CNode* pfrom, std::string& strCommand, CData
         strMessage = addr.ToString() + boost::lexical_cast<std::string>(sigTime) + vchPubKey + vchPubKey2 + boost::lexical_cast<std::string>(protocolVersion) + donationAddress.ToString() + boost::lexical_cast<std::string>(donationPercentage);
 
         if (protocolVersion < masternodePayments.GetMinMasternodePaymentsProto()) {
-            LogPrintf("dsee - ignoring outdated Masternode %s protocol version %d < %d\n", vin.prevout.hash.ToString(), protocolVersion, masternodePayments.GetMinMasternodePaymentsProto());
+            LogPrint("masternode","dsee - ignoring outdated Masternode %s protocol version %d < %d\n", vin.prevout.hash.ToString(), protocolVersion, masternodePayments.GetMinMasternodePaymentsProto());
```
Comment (Fuzzbawls) [inline src/masternodeman.cpp]: post-merge: This should have been left as `LogPrintf()` as it results in a `Misbehaving` message.

File: src/masternodeman.cpp [hunk 4]
```diff
@@ -897,7 +897,7 @@ void CMasternodeMan::ProcessMessage(CNode* pfrom, std::string& strCommand, CData
         pubkeyScript = GetScriptForDestination(pubkey.GetID());
 
         if (pubkeyScript.size() != 25) {
-            LogPrintf("dsee - pubkey the wrong size\n");
+            LogPrint("masternode","dsee - pubkey the wrong size\n");
```
Comment (Fuzzbawls) [inline src/masternodeman.cpp]: post-merge: This should have been left as `LogPrintf()` as it results in a `Misbehaving` message.

File: src/masternodeman.cpp [hunk 5]
```diff
@@ -906,20 +906,20 @@ void CMasternodeMan::ProcessMessage(CNode* pfrom, std::string& strCommand, CData
         pubkeyScript2 = GetScriptForDestination(pubkey2.GetID());
 
         if (pubkeyScript2.size() != 25) {
-            LogPrintf("dsee - pubkey2 the wrong size\n");
+            LogPrint("masternode","dsee - pubkey2 the wrong size\n");
```
Comment (Fuzzbawls) [inline src/masternodeman.cpp]: post-merge: This should have been left as `LogPrintf()` as it results in a `Misbehaving` message.

File: src/masternodeman.cpp [hunk 6]
```diff
@@ -906,20 +906,20 @@ void CMasternodeMan::ProcessMessage(CNode* pfrom, std::string& strCommand, CData
...
             Misbehaving(pfrom->GetId(), 100);
             return;
         }
 
         if (!vin.scriptSig.empty()) {
-            LogPrintf("dsee - Ignore Not Empty ScriptSig %s\n", vin.prevout.hash.ToString());
+            LogPrint("masternode","dsee - Ignore Not Empty ScriptSig %s\n", vin.prevout.hash.ToString());
```
Comment (Fuzzbawls) [inline src/masternodeman.cpp]: post-merge: This should have been left as `LogPrintf()` as it results in a `Misbehaving` message.

File: src/masternodeman.cpp [hunk 7]
```diff
@@ -906,20 +906,20 @@ void CMasternodeMan::ProcessMessage(CNode* pfrom, std::string& strCommand, CData
...
             Misbehaving(pfrom->GetId(), 100);
             return;
         }
 
         std::string errorMessage = "";
         if (!obfuScationSigner.VerifyMessage(pubkey, vchSig, strMessage, errorMessage)) {
-            LogPrintf("dsee - Got bad Masternode address signature\n");
+            LogPrint("masternode","dsee - Got bad Masternode address signature\n");
```
Comment (Fuzzbawls) [inline src/masternodeman.cpp]: post-merge: This should have been left as `LogPrintf()` as it results in a `Misbehaving` message.

File: src/masternodeman.cpp [hunk 8]
```diff
@@ -973,7 +973,7 @@ void CMasternodeMan::ProcessMessage(CNode* pfrom, std::string& strCommand, CData
         // make sure the vout that was signed is related to the transaction that spawned the Masternode
         //  - this is expensive, so it's only done once per Masternode
         if (!obfuScationSigner.IsVinAssociatedWithPubkey(vin, pubkey)) {
-            LogPrintf("dsee - Got mismatched pubkey and vin\n");
+            LogPrint("masternode","dsee - Got mismatched pubkey and vin\n");
```
Comment (Fuzzbawls) [inline src/masternodeman.cpp]: post-merge: This should have been left as `LogPrintf()` as it results in a `Misbehaving` message.

File: src/masternodeman.cpp [hunk 9]
```diff
@@ -999,7 +999,7 @@ void CMasternodeMan::ProcessMessage(CNode* pfrom, std::string& strCommand, CData
 
         if (fAcceptable) {
             if (GetInputAge(vin) < MASTERNODE_MIN_CONFIRMATIONS) {
-                LogPrintf("dsee - Input must have least %d confirmations\n", MASTERNODE_MIN_CONFIRMATIONS);
+                LogPrint("masternode","dsee - Input must have least %d confirmations\n", MASTERNODE_MIN_CONFIRMATIONS);
```
Comment (Fuzzbawls) [inline src/masternodeman.cpp]: post-merge: This should have been left as `LogPrintf()` as it results in a `Misbehaving` message.

File: src/masternodeman.cpp [hunk 10]
```diff
@@ -1070,16 +1070,16 @@ void CMasternodeMan::ProcessMessage(CNode* pfrom, std::string& strCommand, CData
...
         if (sigTime > GetAdjustedTime() + 60 * 60) {
-            LogPrintf("dseep - Signature rejected, too far into the future %s\n", vin.prevout.hash.ToString());
+            LogPrint("masternode","dseep - Signature rejected, too far into the future %s\n", vin.prevout.hash.ToString());
             Misbehaving(pfrom->GetId(), 1);
             return;
         }
 
         if (sigTime <= GetAdjustedTime() - 60 * 60) {
-            LogPrintf("dseep - Signature rejected, too far into the past %s - %d %d \n", vin.prevout.hash.ToString(), sigTime, GetAdjustedTime());
+            LogPrint("masternode","dseep - Signature rejected, too far into the past %s - %d %d \n", vin.prevout.hash.ToString(), sigTime, GetAdjustedTime());
```
Comment (Fuzzbawls) [inline src/masternodeman.cpp]: post-merge: This should have been left as `LogPrintf()` as it results in a `Misbehaving` message.

File: src/masternode-budget.cpp [hunk 11]
```diff
@@ -1001,7 +1001,7 @@ void CBudgetManager::ProcessMessage(CNode* pfrom, std::string& strCommand, CData
         if (Params().NetworkID() == CBaseChainParams::MAIN) {
             if (nProp == 0) {
                 if (pfrom->HasFulfilledRequest("mnvs")) {
-                    LogPrintf("mnvs - peer already asked me for the list\n");
+                    LogPrint("masternode","mnvs - peer already asked me for the list\n");
```
Comment (Fuzzbawls) [inline src/masternode-budget.cpp]: post-merge: This should have been left as `LogPrintf()` as it results in a `Misbehaving` message.

File: src/masternode-budget.cpp [hunk 12]
```diff
@@ -1141,7 +1141,7 @@ void CBudgetManager::ProcessMessage(CNode* pfrom, std::string& strCommand, CData
 
         mapSeenFinalizedBudgetVotes.insert(make_pair(vote.GetHash(), vote));
         if (!vote.SignatureValid(true)) {
-            LogPrintf("fbvote - signature invalid\n");
+            LogPrint("masternode","fbvote - signature invalid\n");
```
Comment (Fuzzbawls) [inline src/masternode-budget.cpp]: post-merge: This should have been left as `LogPrintf()` as it results in a `Misbehaving` message.

File: src/masternode-payments.cpp [hunk 13]
```diff
@@ -669,7 +669,7 @@ bool CMasternodePaymentWinner::IsValid(CNode* pnode, std::string& strError)
         // We don't want to print all of these messages, or punish them unless they're way off
         if (n > MNPAYMENTS_SIGNATURES_TOTAL * 2) {
             strError = strprintf("Masternode not in the top %d (%d)", MNPAYMENTS_SIGNATURES_TOTAL * 2, n);
-            LogPrintf("CMasternodePaymentWinner::IsValid - %s\n", strError);
+            LogPrint("masternode","CMasternodePaymentWinner::IsValid - %s\n", strError);
```
Comment (Fuzzbawls) [inline src/masternode-payments.cpp]: post-merge: This should have been left as `LogPrintf()` as it results in a `Misbehaving` message.

**Pull request merged**